TOOLS/dylib_unhell.py

```diff
@@ -41,83 +41,106 @@ def otool(objfile, rapths):
 
 def get_rapths(objfile):
     rpaths = []
-    command = "otool -l '%s' | grep -A2 LC_RPATH | grep path" % objfile
-    pathRe = re.compile(r"^\s*path (.*) \(offset \d*\)$")
+    command = f"otool -l '{objfile}' | grep -A2 LC_RPATH | grep path"
+    path_re = re.compile(r"^\s*path (.*) \(offset \d*\)$")
 
     try:
-        result = subprocess.check_output(command, shell = True, universal_newlines=True)
-    except:
+        result = subprocess.check_output(command, shell=True, universal_newlines=True)
+    except Exception:
         return rpaths
 
     for line in result.splitlines():
-        line_clean = pathRe.search(line).group(1).strip()
+        line_clean = path_re.search(line).group(1).strip()
         # resolve @loader_path
-        if line_clean.startswith('@loader_path/'):
-            line_clean = line_clean[len('@loader_path/'):]
-            line_clean = os.path.normpath(os.path.join(os.path.dirname(objfile), line_clean))
+        if line_clean.startswith("@loader_path/"):
+            line_clean = line_clean[len("@loader_path/"):]
+            line_clean = os.path.join(os.path.dirname(objfile), line_clean)
+            line_clean = os.path.normpath(line_clean)
         rpaths.append(line_clean)
 
     return rpaths
 
 def get_rpaths_dev_tools(binary):
-    command = "otool -l '%s' | grep -A2 LC_RPATH | grep path | grep \"Xcode\\|CommandLineTools\"" % binary
+    command = (
+        f"otool -l '{binary}' | grep -A2 LC_RPATH | grep path | "
+        'grep "Xcode\\|CommandLineTools"'
+    )
```

Comment:
i wonder about those parentheses? why did this string became a tuple(?)?

in other cases those stayed a string or a list (with square brackets). we should be consistent at least.

Reply:
Python will merge two string literal without an operator between them, similar to the way C does. The reasons for the brackets is to tell python that the strings to be merged can go across multiple lines. It will only be considered a tuple if there are commas separating the values.

Reply:
thanks for the clarification.